javascript.md

```diff
@@ -0,0 +1,157 @@
+# JavaScript
+
+# Naming
+
+* Variable declarations should always be made using var to not declare them as global variables.
+
+  # Bad
+  function sum(x, y) {
+    result = x + y;
+    return result;
+  }
+
+  # Good
+  function sum(x, y) {
+    var result = x + y;
+    return result;
+  }
+
+
+* use camelCase for variables and functions.
+
+* Constants (like PI) are written in UPPERCASE.
+
+* JavaScript files should have a .js extension.
+
+* Constructors intended for use UpperCamelCase.
+
+
+# Syntax
+
+* Always end a simple statement with a semicolon.
+
+* In case of object use colon plus one space between each property and its value.
+
+* In case of keys in object use quotes around string values, not around numeric values.
+
+* Do not add a comma after the last property-value pair of an object.
+
+* For understandability try to declare all variables at the top.
+
+* To check both data type and values use '==='.
+
+  # Bad
+  var a = "5";
+  var b = 5;
+
+  if (a==b && typeof(a) == typeof(b)) {
+    // do something
+  }
+  
+  # Good
+  var a = "5";
+  var b = 5;
+
+  if (a===b) {
+    // do something  
+  }
+
+# Programming
+
+* Use 2 space indentation.
+
+* Always put spaces around operators ( = + - * / ), and after commas.
+
+* Always put the opening brace on the same line as the previous statement.
```

Comment:
Any one can be done but consistency should be maintained throughout the project.